src/huntsman/pocs/observatory.py

```diff
@@ -310,66 +320,79 @@ def take_dark_fields(self,
         if not isinstance(n_darks, list):
             n_darks = listify(n_darks) * len(exptimes)
 
+        if set_from_config:
+            dark_config = self.get_config('calibration', default=dict())
+            exptimes = dark_config["darks"]["exposure_time"]
+            n_darks = dark_config["darks"]["n_darks"]
+
         # Loop over cameras.
-        for exptime, num_darks in zip(exptimes, n_darks):
+        if len(exptimes) > 0:
```

Comment:
@JAAlvarado-Montes how about have a if statement before this block instead so you don't have to indent this entire thing. I think you want this to error as well.

```suggestion
        if len(exptimes) == 0:
            raise PanError("No exposure times were provided. No dark images were taken.")
```